src/OrchardCore.Modules/OrchardCore.AdminMenu/Controllers/MenuController.cs

```diff
@@ -223,6 +227,40 @@ public async Task<IActionResult> Delete(string id)
             return RedirectToAction(nameof(List));
         }
 
+        [HttpPost, ActionName("List")]
+        [FormValueRequired("submit.BulkAction")]
+        public async Task<ActionResult> IndexPost(ViewModels.ContentOptions options, IEnumerable<string> itemIds)
+        {
+            if (itemIds?.Count() > 0)
+            {
+                var adminMenuList = (await _adminMenuService.GetAdminMenuListAsync()).AdminMenu;
+                var checkedContentItems = adminMenuList.Where(x => itemIds.Contains(x.Id));
+                switch (options.BulkAction)
+                {
+                    case ContentsBulkAction.None:
+                        break;
+                    case ContentsBulkAction.Remove:
+                        foreach (var item in checkedContentItems)
+                        {
+                            if (!await _authorizationService.AuthorizeAsync(User, Permissions.ManageAdminMenu, item))
+                            {
+                                _notifier.Warning(H["Couldn't remove selected admin menus."]);
+                                return Forbid();
```

Comment:
this seems like it could be done once and without sending the item through for permissions, because the `ManageAdminMenu` permission doesn't check the resource.

It would be odd if one returned not authorized, but it had deleted 3 first.

Reply:
Hmm this is flaky but we do this everywhere ; maybe it's worth a different PR.

Reply:
In fact it should at least return a message about which menu it did not successfully delete.

Reply:
Where is everywhere? 

I can't see anywhere else that we are doing this except for contentitems, which makes sense because they can be authorized on an individual level.

Reply:
I did not copy this from the ContentItems controller. But I see what you mean. Since this is a single document in the database it's worth nothing.